src/job.rs

```diff
@@ -382,12 +383,13 @@ impl JobLocked {
     /// sched.add(job)
     /// tokio::spawn(sched.start());
     /// ```
-    pub fn new<T>(schedule: &str, run: T) -> Result<Self, Box<dyn std::error::Error>>
+    pub fn new<S, T>(schedule: S, run: T) -> Result<Self, Box<dyn std::error::Error>>
     where
         T: 'static,
         T: FnMut(Uuid, JobsSchedulerLocked) + Send + Sync,
+        S: TryInto<Schedule, Error=cron::error::Error>
```

Comment:
`Error=cron::error::Error` won't work here. The function needs to take an `E: std::error::Error` type parameter in order to handle both the `str` parameter, as well as the blanket impl for `TryFrom<U: Into<T>> for T`, which has `std::convert::Infallible` as the associated `Error` type. (https://doc.rust-lang.org/std/convert/trait.TryFrom.html#impl-TryFrom%3CU%3E)